debug/Debug.js

```diff
@@ -0,0 +1,130 @@
+function Debug(scene) {
+    // CHARTS
+    // create charts div
+    const chartDiv = document.createElement('div');
+    chartDiv.id = 'chart-div';
+    chartDiv.style = 'position: absolute; bottom: 0; left: 0; width: 100vw; height: 20rem; background-color: white';
+
+    scene.viewerDiv.appendChild(chartDiv);
+
+    // line graph for nb elements
+    const nbObjectsCanvas = document.createElement('canvas');
+    nbObjectsCanvas.heigth = '20rem';
+    nbObjectsCanvas.width = '40rem';
+    nbObjectsCanvas.id = 'nb-objects';
+    chartDiv.appendChild(nbObjectsCanvas);
+
+    const timestamp = Date.now();
+    const nbObjectsDataset = { label: 'Number of object in Scene', data: [{ x: 0, y: 0 }] };
+    const nbObjectsChartLabel = ['0s'];
+    const nbObjectsChart = new Chart('nb-objects', {
+        type: 'line',
+        data: {
+            labels: nbObjectsChartLabel,
+            datasets: [nbObjectsDataset],
+        },
+        options: {
+            animation: { duration: 10 },
+            scales: {
+                yAxes: [{
+                    display: true,
+                    ticks: {
+                        suggestedMin: 0, // minimum will be 0, unless there is a lower value.
+                    },
+                }],
+            },
+        },
+    });
+
+    function updateNbObjectsChart() {
+        function countElem(node) {
+            if (!node) {
+                return 0;
+            }
+            let count = 1; // this node
+            if (node.children) {
+                for (const child of node.children) {
+                    count += countElem(child);
+                }
+            }
+            return count;
+        }
+
+        const timeInS = Math.floor((Date.now() - timestamp) / 1000);
+
+        const limit = 50;
+        nbObjectsChartLabel.push(`${timeInS}s`);
+        if (nbObjectsChartLabel.length > limit) {
+            nbObjectsChartLabel.shift();
+        }
+        nbObjectsDataset.data.push({ x: timeInS, y: countElem(scene.gfxEngine.scene3D) });
+        if (nbObjectsDataset.data.length > limit) {
+            nbObjectsDataset.data.shift();
+        }
+        nbObjectsChart.update();
+    }
+
+    // hook that to scene.update
+    const oldUpdate = Object.getPrototypeOf(scene).update.bind(scene);
+    scene.update = function debugUpdate() {
+        oldUpdate();
+        updateNbObjectsChart();
+    };
+
+    // DEBUG CONTROLS
+    const gui = new dat.GUI();
+
+    const state = {
+        showOutline: false,
+        wireframe: false,
+        displayCharts: true,
+    };
+
+    // charts
+    gui.add(state, 'displayCharts').name('Display charts').onChange((newValue) => {
+        if (newValue) {
+            chartDiv.style.display = 'flex';
+        } else {
+            chartDiv.style.display = 'none';
+        }
+    });
+
+    // tiles outline
+    gui.add(state, 'showOutline').name('Show tiles outline').onChange((newValue) => {
+        var cO = function c0(object) {
```

Comment:
> how do you use the material to have one layer in wireframe (tiles by example) and other layer in solid (building by example)?

At the moment, the current behaviour of the checkbox is to activate it for all layers. It would always be time to change the location of this boolean when/if someone implements a wireframe checkbox for each layers. Let's be concerned about current behaviour and not future/hypothetic ones.

> Factorize debug GUI in Examples/Gui 

Debug is not examples right? It's an internal tools imo.

> encapsulate set wireframe and outline in api

For this, I would say it's outside the scope of this PR. This PR is concerned with debug tools to be used internally by us. We don't know yet if this belongs to the public API. Let's wait to see if there is a need (it's always either to add things to an API, but very difficult to remove them once it's there). It won't be difficult to add it in the future anyway.

> move GUI debug in folder in original bottom layers.

I don't understand what you're asking.

> It seems to me that the prototype modification also changes BasicMaterial

Nope, why would it?